src/feature.js

```diff
@@ -115,18 +72,17 @@ class FxABrowserFeature {
   _defaultAvatar() {
     if (this._avatar) {
       this._avatar = null;
-      // TODO: Render first character of email instead
       browser.browserAction.setIcon({ path: "icons/avatar.svg" });
     }
   }
 
   _userAvatar(url) {
     console.log("FxABrowserFeature::_userAvatar");
-    if (this._avatar && this._avatarUrl === url) {
+    if (this._avatar === url) {
```

Comment:
There was something confusing going on here, we had both `this._avatar` and `this._avatarUrl`, but only one of them was being used (and only one is needed) for the short-circuit behaviour. So I got rid of the one that wasn't being used.